mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx

```diff
@@ -44,19 +45,41 @@ export const GoalMetricItem = (props: SingleGoalMetricProps) => {
     ? `Done date ${DateUtils.getShortISODateValue(props.metric.doneDate)}`
     : "Not finished yet...";
 
+  /**
+   * Set metric not completed
+   */
+  const onOk = () => {
+    props.updateMetric({...props.metric, isDone: !props.metric.isDone, doneDate: new Date()});
+  };
+
   return (
     <div
       key={props.metric.uuid}
       className={styles.singularMetric}
     >
       <HorizontalContainer className={styles.horizontalContainer}>
-        <Checkbox
-          isEditable={props.isEditable}
-          isDefaultChecked={props.metric.isDone}
-          className={styles.checkbox}
-          // TODO #487: check logic about update doneDate. What if I accidentally clicked checkbox few times
-          onChange={(isDone) => props.updateMetric({...props.metric, isDone, doneDate: new Date()})}
-        />
+        {!props.metric.isDone ?
+          <Checkbox
+            isEditable={props.isEditable}
+            isDefaultChecked={props.metric.isDone}
+            className={styles.checkbox}
+            onChange={(isDone) => props.updateMetric({...props.metric, isDone, doneDate: new Date()})}
+          /> :
+          <Confirm
```

Comment:
Does it possible to render only have one checkbox with nu ternnarian operator? Will it look better?

Reply:
I think in this case this is the most suitable solution if you do not change the modal